homeassistant/components/cert_expiry/sensor.py

```diff
@@ -71,12 +74,23 @@ def __init__(self, sensor_name, server_name, server_port):
         self._state = None
         self._available = False
         self._valid = False
+        self._retry_attempts = 0
 
     @property
     def name(self):
         """Return the name of the sensor."""
         return self._name
 
+    @property
+    def next_check(self):
+        """Return the timestamp of the next update retry attempt ."""
+        return dt_util.utcnow() + timedelta(seconds=self.retry_delay)
+
+    @property
+    def retry_delay(self):
+        """Return the retry delay in seconds."""
+        return int(min(2 ** (self._retry_attempts - 1) * 30, 3600))
+
```

Comment:
I'm personally not a fan of such retry logic if that is just simple fix with `SCAN_INTERVAL = timedelta(30min)` and trow an Retry error on config entry setup if the server is not available. Just 2 lines that reduce the complexity of this PR and remove - 60 lines

Reply:
Simpler is definitely better. However, some (but not all) users monitoring the cert of the `http` interface still encounter this on every startup. Having the sensor unavailable for even 30min on startup is a long time. And a 30min interval is far too often for properly validating certs. I still like the exception-based retry mechanism for this use case and a 12h interval otherwise.

With this retry in place, the startup delays based on `EVENT_HOMEASSISTANT_START` can probably be removed to make things a bit simpler again.

What do you think?

Reply:
Okay, I now see the `ConfigEntryNotReady` exception you're referencing. I think that should handle the retry logic on startup. 👍